bin/ci/patch_plugin_gemfile_lock

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env python3
+
+import os
+import re
+
+with open('Gemfile.lock') as f:
+  gemfile_lock = f.read()
+
+gemfile_lock_patched = re.compile("GIT\n  remote: %s\n  revision: .+\n  branch: .+\n" %(os.environ['GITHUB_REPOSITORY'])).sub('PATH\n  remote: .\n', gemfile_lock)
```

Comment:
NOTE these GitHub actions env bars don't with with cross-repo but this is no different from the existing before_install checks so I left it for now

Reply:
Do we need to log the Gemfile.lock before and after patching?  Or maybe just the diff?  I was hoping to see what this looks like in the CI log.

Reply:
There's a typo in your comment... I'm not sure what you're saying.

Reply:
Fixed, it was supposed to be `env vars`, basically when running cross-repo for the oparin branch the `GITHUB_BASE_REF` is still `master` because the PR is created in the cross-repo repository

Reply:
ah... shucks... I just asked below if we can test it with cross repo to see how it works.  I guess you can do the old fashioned way and just create a WIP plugin PR to use this code to verify it works.

Reply:
Yeah I just committed a `Gemfile.lock.release` to bypass the need to copy it from core

Reply:
> Do we need to log the Gemfile.lock before and after patching? Or maybe just the diff? I was hoping to see what this looks like in the CI log.

Gemfile.lock is already logged in CI.

Reply:
I was more interested in seeing what it changed and less about the full contents.